### DISCUSSION TITLE
Testimonial how learn go with tests influenced my career

### DISCUSSION THREAD

**Discussion by dm1tr1yvovk** [quii/learn-go-with-tests#828] [Show and tell]
For the first time, I was learning it in the autumn of 2021
I was trying to learn go during this period and stuck on this resource.

The lessons gave me this sense of quality code that nothing else did.
They have developed intuition if the code is testable and maintainable.

And this feeling goes far beyond golang.
I've written software in python, nodejs, some infrastructure scripts, terraform, reading groovy.

TDD approach with implementation in such detailed lessons gave me discipline and a way to orginise development process.
It game me confidence that I can deliver quality software without bugs.

All this accumulated later in years 2022, 2023 2024.
In the latter years, I started doing not Devops(Cloud platrofm) engineering work, but software engineering tasks.

Those developed "sences" helped me in code review where I can smell that code is bad.

I am sure that best is yet to come in my career, but I am confident that this background would remain solid no matter what, no matter what technology I would be using.

How I was working on the resource?
I was working early in the morning, retyping code, yes, that simple.
and later on trying to dig dipper into what is going on there.

but it gave results, as I see!
at that moment available, probably I've retyped all lessons)

I am super grateful for people created those lessons and maintaining them!

thank you @quii and all other contributors!

**Comment by quii** [comment]
Brilliant to hear this, thanks @worldcompass
